### PR TITLE
#714 add SemEval 16 dataset

### PR DESCRIPTION
**Pull request by ZizhenWang**
#714

### CODE REVIEW

File: matchzoo/datasets/cqa_ql_16/load_data.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+) -> typing.Union[matchzoo.DataPack, tuple]:
+    """
+    Load CQA-QL-16 data.
+
+    :param stage: One of `train`, `dev`, and `test`.
+        (default: `train`)
+    :param task: Could be one of `ranking`, `classification` or a
+        :class:`matchzoo.engine.BaseTask` instance. (default: `ranking`)
+    :param target_label: If `ranking`, choose one of classification
+        label as the positive label. (default: `Good`)
```
Comment (uduse) [inline matchzoo/datasets/cqa_ql_16/load_data.py]: line 22 says `PerfectMatch`.

File: matchzoo/datasets/cqa_ql_16/load_data.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    match_type: str = 'question',
+    method: str = 'both',
+) -> typing.Union[matchzoo.DataPack, tuple]:
+    """
+    Load CQA-QL-16 data.
+
+    :param stage: One of `train`, `dev`, and `test`.
+        (default: `train`)
+    :param task: Could be one of `ranking`, `classification` or a
+        :class:`matchzoo.engine.BaseTask` instance. (default: `ranking`)
```
Comment (uduse) [inline matchzoo/datasets/cqa_ql_16/load_data.py]: line 21 says `classification`

File: matchzoo/datasets/cqa_ql_16/load_data.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    :param target_label: If `ranking`, choose one of classification
+        label as the positive label. (default: `Good`)
+    :param return_classes: `True` to return classes for classification
+        task, `False` otherwise.
+    :param match_type: Matching text types. One of `question`,
+        `answer`, and `external_answer`. (default: `question`)
+    :param method: Train data use method. One of `part1`, `part2`,
```
Comment (uduse) [inline matchzoo/datasets/cqa_ql_16/load_data.py]:
```suggestion
    :param mode: Train data use method. One of `part1`, `part2`,
```
We usually use `mode` for this kind of thing.

File: matchzoo/datasets/cqa_ql_16/load_data.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
+"""CQA-QL-16 data loader."""
+
+import typing
+from pathlib import Path
+
+import keras
+import pandas as pd
+from xml.etree.ElementTree import parse
```
Comment (uduse) [inline matchzoo/datasets/cqa_ql_16/load_data.py]: `import xml` may be better.

File: matchzoo/datasets/cqa_ql_16/load_data.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    if method not in ('part1', 'part2', 'both'):
+        raise ValueError(f"{method} is not a valid method."
+                         f"Must be one of `part1`, `part2`, `both`.")
+
+    data_root = _download_data(stage)
+    data_pack = _read_data(data_root, stage, match_type, method)
+
+    if task == 'ranking':
+        task = matchzoo.tasks.Ranking()
+    if task == 'classification':
```
Comment (bwanglzu) [inline matchzoo/datasets/cqa_ql_16/load_data.py]:
```python
elif task == 'classification':
    # do sometihng
else raise ValueError:
    # error message
```    

and remove line 92-94

File: matchzoo/datasets/cqa_ql_16/load_data.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
+"""CQA-QL-16 data loader."""
+
+import typing
+from pathlib import Path
+
+import xml
```
Comment (bwanglzu) [inline matchzoo/datasets/cqa_ql_16/load_data.py]: `xml` is python standard library

**Pull request merged**